platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/streams/SerializableDataInputStream.java

```diff
@@ -586,4 +580,28 @@ private static <T extends SelfSerializable> T registryConstructor(final long cla
         }
         return rc;
     }
+
+    /**
+     * Reads a PBJ record from the stream.
+     *
+     * @param codec the codec to use to parse the record
+     * @param <T>   the type of the record
+     * @return the parsed record
+     * @throws IOException if an IO error occurs
+     */
+    public @NonNull <T extends Record> T readPbjRecord(@NonNull final Codec<T> codec) throws IOException {
+        final int size = readInt();
+        readableSequentialData.limit(readableSequentialData.position() + size);
+        try {
+            return codec.parse(readableSequentialData);
```

Comment:
Which code is responsible for resetting the limit on the stream? Or the caller is supposed to close the stream after reading a single record from it? This isn't clear from the javadoc, and there's not a `finally{}` block that would undo the limit (or close the stream, for that matter.)

Reply:
The limit is set every time this method is called and the stream is never used outside of this method. So no limit reset is necessary. The stream should not be closed because this assumes that there will be a continuous stream from which you can read any number of PBJ records.

Reply:
So if you call this method once, it reads an integer with the size of the record and then sets a limit on the stream and reads this record.

Suppose you then call this method the second time to read another record from the same stream. The very first call to readInt() will fail now because it will hit a limit set by a previous call to this method.

How is this supposed to be handled?

Reply:
the call to `readInt` does not go through the `readableSequentialData`, so it is not affected by the limit

Reply:
Oh, I didn't realize that. Looks good then. Thanks for the clarification!